src/test/scala/io/citrine/lolo/util/LoloPyDataLoaderTest.scala

```diff
@@ -0,0 +1,111 @@
+    val data = LoloPyDataLoader.get1DArray(getData, getFloat = true, bigEndian = false).asInstanceOf[Seq[Double]]
+    assert(data.length == 16)
+    assert(Math.abs(data(1) - 1) < 1e-6, "Array read badly")
+  }
+
+  @Test
+  def test1DArrayInteger(): Unit = {
+    val data = LoloPyDataLoader.get1DArray(getData, getFloat = false, bigEndian = false).asInstanceOf[Seq[Int]]
+    assert(data.length == 32)
+    assert(data.take(3).sum == 0) // First 3 values are all 0
+    assert(data(3) == 1072693248) // Fourth is a big one
+  }
+
+  @Test
+  def testZipping(): Unit = {
+    val X = LoloPyDataLoader.getFeatureArray(getData, 4, bigEndian = false)
+    val y = LoloPyDataLoader.get1DArray(getData, getFloat = true, bigEndian = false).asInstanceOf[Seq[Double]]
+    val trainData = LoloPyDataLoader.zipTrainingData(X, y)
+    assert(trainData.length == 4)
+    val (x_i, y_i: Double) = trainData.head
+    assert(x_i.length == 4)
+    assert(Math.abs(x_i(1) - 1) < 1e-6)
+    assert(Math.abs(y_i - 0) < 1e-6)
+  }
+
+  @Test
+  def testGetRegression(): Unit = {
+    // Generate an RF model
+    val trainingData = TestUtils.binTrainingData(
+      TestUtils.generateTrainingData(32, 12, noise = 0.1, function = Friedman.friedmanSilverman),
+      inputBins = Seq((0, 8))
+    )
+
+    val RFMeta = RandomForest.apply(numTrees = trainingData.length * 2)
+      .train(trainingData)
+    val RF = RFMeta.getModel()
+
+    // Generate the prediction results object
+    val results = RF.transform(trainingData.map(_._1))
+
+    // Get the expected results
+    val means = results.getExpected().asInstanceOf[Seq[Double]]
+    val sigma = results.getUncertainty().get.asInstanceOf[Seq[Double]]
+
+    // Get the results as a byte array, and convert them back
+    val reproMeans = LoloPyDataLoader.get1DArray(
+      LoloPyDataLoader.getRegressionExpected(results), getFloat = true, bigEndian = false).asInstanceOf[Seq[Double]]
+    assert(reproMeans.zip(means).map(x => Math.abs(x._1 - x._2)).sum < 1e-6)
+
+    // Get the uncertainty as a byte array, and convert them back
+    val reproSigma = LoloPyDataLoader.get1DArray(
+      LoloPyDataLoader.getRegressionUncertainty(results), getFloat = true, bigEndian = false).asInstanceOf[Seq[Double]]
+    assert(reproSigma.zip(sigma).map(x => Math.abs(x._1 - x._2)).sum < 1e-6)
+  }
+
+  @Test
+  def testGetClassification(): Unit = {
+    // Make a classification model
+    val trainingData = TestUtils.binTrainingData(
+      TestUtils.generateTrainingData(32, 12, noise = 0.1, function = Friedman.friedmanSilverman),
+      inputBins = Seq((0, 8)), responseBins = Some(8)
+    )
+    val RFMeta = new RandomForest(numTrees = trainingData.length * 2)
+      .train(trainingData.map(x => (x._1, Integer.parseInt(x._2.toString))))
+    val RF = RFMeta.getModel()
+    val nClasses : Int = trainingData.map(x => Integer.parseInt(x._2.toString)).max + 1
+
+    // Get the base results
+    val results = RF.transform(trainingData.map(_._1))
```

Comment:
Why do the results need to come from a model?  Couldn't you just generate some random means and probs and then round-robin them through?